mat64/io.go

```diff
@@ -81,18 +87,24 @@ func (m *Dense) UnmarshalBinary(data []byte) error {
 		panic("mat64: unmarshal into non-zero matrix")
 	}
 
+	var tmp [8]byte
 	buf := bytes.NewReader(data)
-	var rows int64
-	err := binary.Read(buf, binary.LittleEndian, &rows)
+	_, err := buf.Read(tmp[:])
```

Comment:
You can do the same thing for writes, but how will this play with your streaming PR?

Make the change to the writes as well, then LGTM and we can sort out the streaming API interaction later.